src/test/java/org/swisspush/redisques/QueueStatsServiceTest.java

```diff
@@ -0,0 +1,191 @@
+            public int limit(Object ctx) {
+                return 10;
+            }
+
+            @Override
+            public String filter(Object ctx) {
+                return "*";
+            }
+
+            @Override
+            public void onQueueStatistics(java.util.List<QueueStatsService.Queue> queues, Object ctx) {
+                // Assert the dequeue stats were attached
+                testContext.assertTrue(queues.get(0).getSize() >0);
+                async.complete();
+            }
+
+            @Override
+            public void onError(Throwable ex, Object ctx) {
+                testContext.fail(ex);
+            }
+        };
+
+        // Mock fetchQueueNamesAndSize to return a valid list of queues
+        Mockito.doAnswer(invocation -> {
+            BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>> callback = invocation.getArgument(1);
+            QueueStatsService.GetQueueStatsRequest<Object> request = invocation.getArgument(0);
+            request.queues = Collections.singletonList(new QueueStatsService.Queue("testQueue", 1));
+            callback.accept(null, request);
+            return null;
+        }).when(queueStatsService).fetchQueueNamesAndSize(Mockito.any(), Mockito.any());
+
+        // Mock fetchRetryDetails to do nothing and call the next step
+        Mockito.doAnswer(invocation -> {
+            BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>> callback = invocation.getArgument(1);
+            QueueStatsService.GetQueueStatsRequest<Object> request = invocation.getArgument(0);
+            callback.accept(null, request);
+            return null;
+        }).when(queueStatsService).fetchRetryDetails(Mockito.any(), Mockito.any());
+
+        // Call getQueueStats to trigger the flow
+        queueStatsService.getQueueStats(mockContext, mentor);
+
+        // Use ArgumentCaptor to capture the arguments passed to attachDequeueStats
+        ArgumentCaptor<QueueStatsService.GetQueueStatsRequest<Object>> requestCaptor = ArgumentCaptor.forClass(QueueStatsService.GetQueueStatsRequest.class);
+        ArgumentCaptor<BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>>> consumerCaptor = ArgumentCaptor.forClass(BiConsumer.class);
+
+        // Verify that attachDequeueStats was called with the correct arguments
+        Mockito.verify(queueStatsService).attachDequeueStats(requestCaptor.capture(), consumerCaptor.capture());
+
+        // Assert that the arguments are not null and match expectations
+        testContext.assertNotNull(requestCaptor.getValue());
+        testContext.assertNotNull(consumerCaptor.getValue());
+    }
+
+    @Test
+    public void testDequeueStatsNotCalledWhenIntervalIsZeroOrNegative(TestContext testContext) {
+        // Mock the config so that getDequeueStatisticReportIntervalSec returns 0 or a negative value
+        Mockito.when(config.getDequeueStatisticReportIntervalSec()).thenReturn(0); // Value equal to 0
```

Comment:
Maybe introduce a new helper method in `RedisquesConfiguration` like `dequeueStatsEnabled` which internally uses `getDequeueStatisticReportIntervalSec() > 0`. It makes the code easier to read. Also replace the current usages of `getDequeueStatisticReportIntervalSec() > 0` with this helper method.